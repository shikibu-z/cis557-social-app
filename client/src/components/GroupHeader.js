/* eslint-disable import/no-extraneous-dependencies */
import React, { useState } from 'react';
import { useHistory } from 'react-router';
import '../stylesheets/GroupHeader.css';
import InviteUser from './InviteUser';

const lib = require('../api/fetchers');

function GroupHeader(props) {
  const id = localStorage.getItem('currUser');
  const { src, group } = props;
  const status = 'Leave';
  const [modalShow, setModalShow] = useState(false);
  const history = useHistory();

  function handleLeave() {
    lib.leaveGroup(group.group.idGroups, { userId: id }).then(() => {
      history.push('/homepage');
    });
  }

  return (
    <div className="groupHeaderWrapper">
      <img src={src} alt="group-img" />
      <div id="groupNameWrapper">
        {group.group.groupName}
      </div>
      <div>
        <button id="inviteBtn" type="button" onClick={() => setModalShow(true)}>Invite</button>
      </div>
      <div id="groupBtnWrapper">
        <button id="joinLeaveBtn" type="submit" onClick={handleLeave}>{status}</button>
      </div>
      <InviteUser
        show={modalShow}
        onHide={() => setModalShow(false)}
        type="Invite"
        group={group.group.idGroups}
      />
    </div>
  );
}

export default GroupHeader;
