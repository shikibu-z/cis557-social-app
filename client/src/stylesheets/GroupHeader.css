.groupHeaderWrapper{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 1% 2% 1% 2%;
}
#groupImgWarapper{
  display: block;
}

#groupNameWrapper{
  display: block;
  font-family: Roboto;
  font-size: 32px;
  font-style: normal;
  font-weight: bold;
}

#groupBtnWrapper{
  display: block;
}

#joinLeaveBtn{
  width: 150px;
  height: 40px;
  border-radius: 100px;
  background-color: #F8B114;
  color: white;
  border: none;
  font-family: Roboto;
  font-size: 22px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0px;
  text-align: center;
}

#inviteBtn {
  width: 150px;
  height: 40px;
  border-radius: 100px;
  background-color: #F8B114;
  color: white;
  border: none;
  font-family: Roboto;
  font-size: 22px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0px;
  text-align: center;
}

.groupHeaderWrapper img{
  display: block;
  width: 150px;
  height: 150px;
  border-radius: 50%;
  overflow: hidden;
  margin: 1%;
}