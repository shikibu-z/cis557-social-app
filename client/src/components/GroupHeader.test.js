/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import GroupHeader from './GroupHeader';

test('GroupHeader matches snapshot', () => {
  const component = renderer.create(<GroupHeader src="" group={{ group: { groupName: 'test' } }} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
