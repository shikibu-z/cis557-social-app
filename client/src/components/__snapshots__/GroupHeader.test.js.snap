// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupHeader matches snapshot 1`] = `
<div
  className="groupHeaderWrapper"
>
  <img
    alt="group-img"
    src=""
  />
  <div
    id="groupNameWrapper"
  >
    test
  </div>
  <div>
    <button
      id="inviteBtn"
      onClick={[Function]}
      type="button"
    >
      Invite
    </button>
  </div>
  <div
    id="groupBtnWrapper"
  >
    <button
      id="joinLeaveBtn"
      onClick={[Function]}
      type="submit"
    >
      Leave
    </button>
  </div>
</div>
`;
